File: src/main/java/net/hectus/neobb/turn/person_game/block/PTBrownStainedGlass.java
package net.hectus.neobb.turn.person_game.block;

import com.marcpg.libpg.util.MinecraftTime;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.block.BlockTurn;
import net.hectus.neobb.turn.person_game.categorization.DefensiveCategory;
import net.hectus.neobb.turn.person_game.warp.PTVillagerWarp;
import net.hectus.neobb.util.Modifiers;
import org.bukkit.block.Block;

public class PTBrownStainedGlass extends BlockTurn implements DefensiveCategory {
    public PTBrownStainedGlass(NeoPlayer player) { super(player); }
    public PTBrownStainedGlass(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public void apply() {
        super.apply();
        player.heal(player.game.warp() instanceof PTVillagerWarp ? 2 : 5);
    }

    @Override
    public void applyDefense() {
        player.addModifier(Modifiers.P_DEFAULT_DEFENDED);
    }

    @Override
    public boolean unusable() {
        return player.game.time() == MinecraftTime.MIDNIGHT;
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/person_game/block/PTBrownStainedGlass.kt
package net.hectus.neobb.turn.person_game.block

import com.marcpg.libpg.storing.Cord
import com.marcpg.libpg.util.MinecraftTime
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.turn.default_game.block.BlockTurn
import net.hectus.neobb.turn.person_game.categorization.DefensiveCategory
import net.hectus.neobb.turn.person_game.warp.PTVillagerWarp
import net.hectus.neobb.util.Modifiers
import org.bukkit.block.Block

class PTBrownStainedGlass(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), DefensiveCategory {
    override fun apply() {
        player!!.heal(if (player.game.warp is PTVillagerWarp) 2.0 else 5.0)
    }

    override fun applyDefense() {
        player!!.addModifier(Modifiers.Player.Default.DEFENDED)
    }

    override fun unusable(): Boolean {
        return player!!.game.time == MinecraftTime.MIDNIGHT
    }
}